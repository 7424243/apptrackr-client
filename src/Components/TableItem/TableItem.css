td {
    text-align: left;
    border-bottom: 1px solid black;
    border-right: 1px solid black;
    padding: 7px;
}

/* Classes */

.last_td {
    border-right: none;
}