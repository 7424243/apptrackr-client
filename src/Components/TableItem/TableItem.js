import React from 'react'
import {Link} from 'react-router-dom'
import { format } from 'date-fns';
import './TableItem.css'

function TableItem(props) {

    return (
        <tbody>
            <tr>
                <td><Link to={`/jobapps/${props.id}`}>{props.job}</Link></td>
                <td>{props.company}</td>
                <td>{props.dateApplied ? format(new Date(props.dateApplied), 'MM/dd/yyy') : null}</td>
                <td>{props.interviewDate ? format(new Date(props.interviewDate), 'MM/dd/yyyy') : null}</td>
                <td className='last_td'>{props.status}</td>
            </tr>
        </tbody>
    )
}

export default TableItem