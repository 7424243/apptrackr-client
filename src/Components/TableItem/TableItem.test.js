import ReactDOM from 'react-dom'
import {BrowserRouter} from 'react-router-dom'
import renderer from 'react-test-renderer'
import TableItem from './TableItem'

describe('TableItem Component', () => {

    const props = {
        id: 1,
        company: 'test',
        dateApplied: 'test',
        interviewDate: 'test',
        status: 'test'
    }

    //smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<BrowserRouter><TableItem {...props}/></BrowserRouter>, div)
        ReactDOM.unmountComponentAtNode(div)
    })

    //snapshot test
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<BrowserRouter><TableItem {...props}/></BrowserRouter>)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
    
})