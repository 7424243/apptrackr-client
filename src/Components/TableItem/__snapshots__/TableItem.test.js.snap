// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableItem Component renders the UI as expected 1`] = `
<tbody>
  <tr>
    <td>
      <a
        href="/jobapps/1"
        onClick={[Function]}
      >
        test
      </a>
    </td>
    <td>
      test
    </td>
    <td>
      03/08/2021
    </td>
    <td>
      03/08/2021
    </td>
    <td
      className="last_td"
    >
      test
    </td>
  </tr>
</tbody>
`;
